bin/strings2android.js

```diff
@@ -43,21 +43,52 @@ function escapeResourceXML( unsafeXMLValue ) {
 	} );
 }
 
+/**
+ * Replacement for complying the TypographyDashes Android lint rule.
+ * Reference: https://android.googlesource.com/platform/tools/base/+/master/lint/libs/lint-checks/src/main/java/com/android/tools/lint/checks/TypographyDetector.java#58
+ *
+ * @param {string} XMLValue input to apply replacement.
+ * @return {string} valid string passing TypographyDashes Android lint rule.
+ */
+function replaceHyphenInNumberRanges( XMLValue ) {
+	return XMLValue.replace( /(\d+\s*)(-)(\s*\d+)/gm, function(
+		match,
+		p1,
+		p2,
+		p3
+	) {
+		// Make sure that if there is no space before digit there isn't
+		// one on the left either -- since we don't want to consider
+		// "1 2 -3" as a range from 2 to 3
+		const isNegativeNumber = p3[ 0 ] !== ' ' && p1[ p1.length - 1 ] === ' ';
+
+		return [ p1, p3 ].join( isNegativeNumber ? p2 : '–' );
+	} );
+}
+
+/**
+ * Replacement for complying the TypographyEllipsis Android lint rule.
+ * Reference: https://android.googlesource.com/platform/tools/base/+/master/lint/libs/lint-checks/src/main/java/com/android/tools/lint/checks/TypographyDetector.java#106
+ *
+ * @param {string} XMLValue input to apply replacements.
+ * @return {string} valid string passing TypographyEllipsis Android lint rule.
+ */
+const ELLIPSIS_PATTERN = /(\.\.\.)/gm;
```

Comment:
Minor: might want to name this to 'THREE_DOTS_PATTERN` since it's not matching an ellipsis character. Similar to here : https://github.com/WordPress/gutenberg/blob/72730e0f326df43e761f5d29759b89a4bec3dd58/packages/eslint-plugin/rules/i18n-ellipsis.js#L11

Reply:
Right, I agree it makes more sense to name it `THREE_DOTS_PATTERN` 👍 .

Reply:
I applied this suggestion in [Rename ellipsis pattern in strings2android.](https://github.com/wordpress-mobile/gutenberg-mobile/pull/4484/commits/6c14c2841ac1dc4f574522e8899f8e77154e0f2c).